pkg/utils/kubernetes/health/health.go

```diff
@@ -315,6 +317,38 @@ func CheckSeed(seed *gardenv1beta1.Seed, identity *gardenv1beta1.Gardener) error
 	return nil
 }
 
+// CheckExtensionObject checks if an extension Object is healthy or not.
+// An extension object is healthy if
+// * Its observed generation is up-to-date
+// * No gardener.cloud/operation is set
+// * No lastError is in the status
+// * A last operation is state succeeded is present
+func CheckExtensionObject(obj extensionsv1alpha1.Object) error {
+	status := obj.GetExtensionStatus()
+	if status.GetObservedGeneration() != obj.GetGeneration() {
+		return fmt.Errorf("observed generation outdated (%d/%d)", status.GetObservedGeneration(), obj.GetGeneration())
+	}
+
+	op, ok := obj.GetAnnotations()[gardencorev1alpha1.GardenerOperation]
+	if ok {
+		return fmt.Errorf("gardener operation %q is still ongoing", op)
+	}
+
+	if lastErr := status.GetLastError(); lastErr != nil {
```

Comment:
Should we move this up (before the `GetAnnotations()` check)? The contract is that the extension only removes the annotation after it has successfully reconciled. If an error occurred the annotation will stay. In this case we would not print the actual error message that would be put into the `lastError` field.
Probably we should move the check whether the annotation is down to the end of the function?